src/Draco.Editor.Web/App.razor

```diff
@@ -0,0 +1,219 @@
+        <option value="Run">Run</option>
+        <option value="CSharp">C#</option>
+        <option value="IL">IL</option>
+    </select>
+</div>
+<div class="editors-container">
+    <MonacoEditor Id="draco-editor" @ref="_dracoEditor" ConstructionOptions="DracoMonacoOptions" OnDidChangeModelContent="CodeChange" OnDidInit="OnInit" />
+    <MonacoEditor Id="output-viewer" @ref="_outputViewer" ConstructionOptions="OutputMonacoOptions" />
+</div>
+
+@code {
+    MonacoEditor? _dracoEditor { get; set; }
+    MonacoEditor? _outputViewer { get; set; }
+    string? _selectedOutputType { get; set; } = "Run";
+    StandaloneEditorConstructionOptions _dracoEditorOptions = new StandaloneEditorConstructionOptions
+        {
+            AutomaticLayout = true,
+            Language = "rust",
+            Value = @"func main() {
+    println(""Hello!"");
+}"
+        };
+
+    StandaloneEditorConstructionOptions _outputEditorOptions = new StandaloneEditorConstructionOptions
+        {
+            AutomaticLayout = true,
+            Language = "",
+            Value = "",
+            ReadOnly = true
+        };
+
+    StandaloneEditorConstructionOptions DracoMonacoOptions(MonacoEditor editor)
+         => _dracoEditorOptions;
+
+    StandaloneEditorConstructionOptions OutputMonacoOptions(MonacoEditor editor)
+        => _outputEditorOptions;
+
+    protected override async Task OnInitializedAsync()
+    {
+        await base.OnInitializedAsync();
+        bool isDarkMode = await JS.InvokeAsync<bool>("isDarkMode");
+        await MonacoEditor.SetTheme(isDarkMode ? "vs-dark" : "vs");
+    }
+
+    async Task OnInit(MonacoEditorBase editor)
+    {
+        string uriString = NavigationManager.Uri;
+        Uri uri = new Uri(uriString);
+        if (!string.IsNullOrWhiteSpace(uri.Fragment))
+        {
+            string strBuffer = new string(uri.Fragment.Skip(1).ToArray());
+            strBuffer = Uri.UnescapeDataString(strBuffer)
+            .Replace('_', '/')
+            .Replace('-', '+')
+            .PadRight(4 * ((strBuffer.Length + 3) / 4), '=');
+            var buffer = Convert.FromBase64String(strBuffer);
+            using (MemoryStream inBuffer = new MemoryStream(buffer))
+            using (DeflateStream gzipStream = new DeflateStream(inBuffer, CompressionMode.Decompress))
+            using (StreamReader outReader = new StreamReader(gzipStream, Encoding.UTF8))
+            {
+                _selectedOutputType = outReader.ReadLine();
+                string code = outReader.ReadToEnd();
+                await _dracoEditor!.SetValue(code);
+            }
+        }
+
+
+        await UpdateOutput();
+    }
+
+    public void OnOutputTypeChange(ChangeEventArgs e)
+    {
+        _selectedOutputType = e.Value.ToString();
+        _ = UpdateOutput(); // awaiting will block UI thread
+    }
+
+    public void CodeChange()
+    {
+        _ = UpdateOutput(); // awaiting will block UI thread.
+    }
+
+    string _codeCache = "";
```

Comment:
Random field, please specify publicity and factor to the top, maybe below the props

Reply:
Copy paste leftover.